packages/bun-usockets/src/context.c

```diff
@@ -757,6 +758,7 @@ struct us_socket_t *us_socket_context_adopt_socket(int ssl, struct us_socket_con
         if (new_s->next) new_s->next->prev = new_s;
     } else {
         us_internal_socket_context_link_socket(context, new_s);
+        us_socket_context_unref(ssl, context);
```

Comment:
can the socket outlive `new_s`? hard to see how this works

Reply:
yeah it can I just need to ref/unref to make sure its alive when unlink/link is called, so we dont free the context if is the last ref to it.